component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -170,6 +180,47 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
         }
     }
 
+    private String getMatchedSubjectAttribute(String certAttributes, String subjectAttributePattern,
+            AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+
+        LdapName ldapDN;
+        try {
+            ldapDN = new LdapName(certAttributes);
+        } catch (InvalidNameException e) {
+            throw new AuthenticationFailedException("error occurred while get the certificate claims", e);
+        }
+        if (subjectPatternCompiled == null) {
+
+            subjectPatternCompiled = Pattern.compile(subjectAttributePattern);
+        }
+        String userNameAttribute = getAuthenticatorConfig().getParameterMap().get(X509CertificateConstants.USERNAME);
+        List<String> matches = new ArrayList<>();
+        for (Rdn distinguishNames : ldapDN.getRdns()) {
+            if (subjectPatternCompiled != null && userNameAttribute.equals(distinguishNames.getType())) {
+                Matcher m = subjectPatternCompiled.matcher(String.valueOf(distinguishNames.getValue()));
+                matches = getMatchedString(m, matches);
+            }
+        }
+        if (matches.isEmpty()) {
+            authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                    X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR_CODE);
+            log.debug(X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR);
+            throw new AuthenticationFailedException(X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR);
+        } else if (matches.size() > 1) {
+            authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                    X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_MULTIPLE_MATCHES_ERROR_CODE);
+            log.debug("More than one value matched with the given regex");
+            throw new AuthenticationFailedException("More than one value matched with the given regex");
```

Comment:
Add more debug data. For example, string which was monitored to see this behavior. Number of matches etc.

Reply:
done